transport/model/interface.go

```diff
@@ -24,7 +25,7 @@ type ProjectService interface {
 }
 
 type ProjectMembershipService interface {
-	CreateInvitation(ctx context.Context, c svcmodel.CreateProjectInvitationInput, authUserID uuid.UUID) (svcmodel.ProjectInvitation, error)
+	CreateInvitation(ctx context.Context, m *background.Manager, c svcmodel.CreateProjectInvitationInput, authUserID uuid.UUID) (svcmodel.ProjectInvitation, error)
```

Comment:
I prefer to pass the manager to the service explicitly rather than including it as part of the service struct. However, I would be happy to hear your opinion on this.

Reply:
Strictly speaking, here we are stating an interface from the transport layer perspective. The fact that invitation creation involves sending an email AND sending it asynchronously shouldn't be transport layer's business at all.

Moreover, should the background manager be a part of service? Probably no; in principle, it only states an interface that there must be somebody who will send an email invitation.
Perhaps here, the most accurate would be to state something like:
```go
type EmailSender interface {
    SendEmailAsync(...)
}
```
So that the service will know that this call will run asynchronously and thus won't take too much time.
Ideally without bothering that it will need a background manager - this will be solved by the `EmailSender` (or in our case `Mailer`) implementation itself. Because it doesn't have to be specifically background manager, right?

Reply:
And so, I'd probably not inject the `TaskManager` in `main`, but rather have it directly as a part of `Resend` struct.

Reply:
(depending on the expected usage etc.; maybe soon it will be used somewhere else and the intended usage is to initialize `TaskManager` in `main` - in this case still, I'd just feed it directly to the `Resend`...)